sdks/java/io/sparkreceiver/src/main/java/org/apache/beam/sdk/io/sparkreceiver/ReadFromSparkReceiverWithOffsetDoFn.java

```diff
@@ -141,15 +145,42 @@ public boolean hasRecords() {
     @Override
     public void start(Receiver<V> sparkReceiver) {
       this.sparkReceiver = sparkReceiver;
-      try {
-        new WrappedSupervisor(
-            sparkReceiver,
-            new SparkConf(),
-            objects -> {
-              V record = (V) objects[0];
-              recordsQueue.offer(record);
+
+      final SerializableFunction<Object[], Void> storeFn =
+          (input) -> {
+            if (input == null) {
               return null;
-            });
+            }
+
+            if (input[0] instanceof ByteBuffer) {
+              final ByteBuffer byteBuffer = ((ByteBuffer) input[0]).asReadOnlyBuffer();
```

Comment:
Why are we just using input[0] ?

Reply:
Added a comment:
`Use only [0] element - data.
 The other elements are not needed because they are related to Spark environment options.`